source/MaterialXCore/Document.cpp

```diff
@@ -463,6 +331,39 @@ bool Document::validate(string* message) const
     return GraphElement::validate(message) && res;
 }
 
+void convertParameterToInput(DocumentPtr doc)
+{
+    // Convert all parameters to be inputs. If needed set them to be "uniform".
+    const StringSet uniformTypes = { FILENAME_TYPE_STRING, STRING_TYPE_STRING };
+    const string PARAMETER_CATEGORY_STRING("parameter");
+    for (ElementPtr e : doc->traverseTree())
+    {
+        InterfaceElementPtr elem = e->asA<InterfaceElement>();
+        if (!elem)
+        {
+            continue;
+        }
+        vector<ElementPtr> children = elem->getChildren();
+        for (ElementPtr child : children)
+        {
+            if (child->getCategory() == PARAMETER_CATEGORY_STRING)
+            {
+                InputPtr newInput = updateChildSubclass<Input>(elem, child);
+                if (uniformTypes.count(child->getAttribute(TypedElement::TYPE_ATTRIBUTE)))
+                {
+                    newInput->setIsUniform(true);
+                }
+                else
+                {
+                    // TODO: Determine based on usage whether to make
+                    // the input a uniform. 
+                    newInput->setIsUniform(false);
```

Comment:
Do we need to set the attribute in this case? The absence of the attribute should indicate that it's not a uniform.

Reply:
Yes, This is left in for now so we can tell what parameters turned into inputs.